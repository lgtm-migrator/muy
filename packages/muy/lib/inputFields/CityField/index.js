import React from "react"
import PropTypes from "prop-types"
import TextField from "@material-ui/core/TextField"
import propTypes from "../propTypes"

const CityField = ({
  autoCapitalize,
  autoComplete,
  autoCorrect,
  id,
  inputMode,
  inputProps,
  label,
  minLength,
  name,
  required,
  ...props
}) => (
  <TextField
    autoComplete={autoComplete}
    label={label}
    id={id}
    inputProps={{
      autoCapitalize,
      autoCorrect,
      "data-name": name,
      inputMode,
      minLength,
      size: 50,
      tabIndex: 0,
      ...inputProps,
    }}
    name={name}
    required={required}
    {...props}
  />
)

export default CityField

CityField.defaultProps = {
  autoCapitalize: "words",
  autoComplete: "address-level-2",
  autoCorrect: "off",
  id: "city",
  inputMode: "latin",
  inputProps: {},
  minLength: 3,
  name: "city",
}

CityField.propTypes = {
  autoCapitalize: propTypes.autoCapitalize,
  autoComplete: PropTypes.string,
  autoCorrect: PropTypes.string,
  id: PropTypes.string,
  inputMode: PropTypes.string,
  inputProps: PropTypes.object,
  label: PropTypes.string,
  minLength: PropTypes.number,
  name: PropTypes.string,
  required: PropTypes.bool,
}
