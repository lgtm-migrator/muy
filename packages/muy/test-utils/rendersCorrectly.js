import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
