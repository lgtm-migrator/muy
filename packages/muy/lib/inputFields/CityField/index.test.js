import React from "react"
import rendersCorrectly from "../../../test-utils/rendersCorrectly"
import CityField from "."

describe("<CityField /> field", () => {
  it("should render correctly with no props", () => {
    rendersCorrectly(<CityField />)
  })
  it("should render correctly with label", () => {
    rendersCorrectly(<CityField label={"City Field Label Text"} />)
  })
})
