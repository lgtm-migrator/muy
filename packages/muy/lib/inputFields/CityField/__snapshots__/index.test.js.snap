// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CityField /> field should render correctly with label 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink={false}
    htmlFor="city"
    id="city-label"
  >
    City Field Label Text
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoCapitalize="words"
      autoComplete="address-level-2"
      autoCorrect="off"
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      data-name="city"
      disabled={false}
      id="city"
      inputMode="latin"
      minLength={3}
      name="city"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      size={50}
      tabIndex={0}
      type="text"
    />
  </div>
</div>
`;

exports[`<CityField /> field should render correctly with no props 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoCapitalize="words"
      autoComplete="address-level-2"
      autoCorrect="off"
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      data-name="city"
      disabled={false}
      id="city"
      inputMode="latin"
      minLength={3}
      name="city"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      size={50}
      tabIndex={0}
      type="text"
    />
  </div>
</div>
`;
